resource_obmcs_core_instance_test.go

```diff
@@ -154,34 +161,109 @@ func (s *ResourceCoreInstanceTestSuite) TestAccResourceCoreInstance_basic() {
 						return err
 					},
 				),
 			},
-			// verify force new by creating an instance with different VNIC details.
+			// Adding create_vnic_details flags with default values should not lead to a change.
 			{
+				ImportState:       true,
+				ImportStateVerify: true,
 				Config: s.Config + `
 				resource "oci_core_instance" "t" {
 					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
 					compartment_id = "${var.compartment_id}"
 					image = "${data.oci_core_images.t.images.0.id}"
 					shape = "VM.Standard1.1"
 					display_name = "-tf-instance"
+					subnet_id = "${oci_core_subnet.t.id}"
 					metadata {
 						ssh_authorized_keys = "${var.ssh_public_key}"
 					}
 					create_vnic_details {
 						subnet_id = "${oci_core_subnet.t.id}"
 						display_name = "-tf-vnic"
+						skip_source_dest_check = false
 						assign_public_ip = true
+					}
+				}
+				data "oci_core_vnic_attachments" "t" {
+					compartment_id = "${var.compartment_id}"
+					instance_id = "${oci_core_instance.t.id}"
+				}
+				data "oci_core_vnic" "t" {
+					vnic_id = "${lookup(data.oci_core_vnic_attachments.t.vnic_attachments[0],"vnic_id")}"
+				}`,
+				ExpectNonEmptyPlan: false,
+				PlanOnly:           true,
+			},
+			// Update create_vnic_details
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config: s.Config + `
+				resource "oci_core_instance" "t" {
+					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
+					compartment_id = "${var.compartment_id}"
+					image = "${data.oci_core_images.t.images.0.id}"
+					shape = "VM.Standard1.1"
+					display_name = "-tf-instance"
+					subnet_id = "${oci_core_subnet.t.id}"
+					metadata {
+						ssh_authorized_keys = "${var.ssh_public_key}"
+					}
+					create_vnic_details {
+						subnet_id = "${oci_core_subnet.t.id}"
+						display_name = "-tf-vnic-2"
+						skip_source_dest_check = true
+						hostname_label = "mytftesthostname"
+					}
+				}
+				data "oci_core_vnic_attachments" "t" {
+					compartment_id = "${var.compartment_id}"
+					instance_id = "${oci_core_instance.t.id}"
+				}
+				data "oci_core_vnic" "t" {
+					vnic_id = "${lookup(data.oci_core_vnic_attachments.t.vnic_attachments[0],"vnic_id")}"
+				}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-tf-instance"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.skip_source_dest_check", "true"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.display_name", "-tf-vnic-2"),
+					func(ts *terraform.State) (err error) {
+						newId, err := fromInstanceState(ts, s.ResourceName, "id")
+						if newId != instanceId {
+							return fmt.Errorf("Expected same instance ocid, got different.")
+						}
+						return err
+					},
+				),
+			},
+			// verify force new by setting non-updateable VNIC details.
```

Comment:
updatable :#

Reply:
https://english.stackexchange.com/questions/56431/updatable-vs-updateable-which-is-correct

Looks like they're both valid (although updatable is more popular).